test/tools/test_tools_time.py

```diff
@@ -47,6 +71,96 @@ def test_validate_format_none():
         time.validate_format(None)
 
 
+def test_get_nick_timezone(db: SopelDB):
+    nick = 'IronMan'
+    assert time.get_nick_timezone(db, nick) is None
+
+    db.set_nick_value(nick, 'timezone', 'Europe/Paris')
+    assert time.get_nick_timezone(db, nick) == 'Europe/Paris'
+
+    db.set_nick_value(nick, 'timezone', 'Invalid_TZ')
+    assert time.get_nick_timezone(db, nick) is None
+
+
+def test_get_channel_timezone(db: SopelDB):
+    channel = '#test'
+    assert time.get_channel_timezone(db, channel) is None
+
+    db.set_channel_value(channel, 'timezone', 'Europe/Paris')
+    assert time.get_channel_timezone(db, channel) == 'Europe/Paris'
+
+    db.set_channel_value(channel, 'timezone', 'Invalid_TZ')
+    assert time.get_channel_timezone(db, channel) is None
+
+
+def test_get_timezone(db: SopelDB, tmpconfig: Config):
+    zone = 'Europe/Paris'
+    zone_nick = 'America/New_York'
+    zone_channel = 'Asia/Tokyo'
+    nick = 'IronMan'
+    channel = '#test'
+
+    assert time.get_timezone(db) is None
+    assert time.get_timezone(db, tmpconfig) == 'UTC', (
+        'Must use default timezone from config'
+    )
+    assert time.get_timezone(db, tmpconfig, zone) == zone
+    assert time.get_timezone(db, tmpconfig, zone, nick, channel) == zone
+
+    # channel tz when no tz and no nick tz (or tz is invalid)
+    db.set_channel_value(channel, 'timezone', zone_channel)
+    assert time.get_timezone(
+        db, tmpconfig, None, nick, channel
+    ) == zone_channel, 'Channel '
+    assert time.get_timezone(
+        db, tmpconfig, 'Invalid/Tz', nick, channel
+    ) == zone_channel
+
+    # nick tz has priority
+    db.set_nick_value(nick, 'timezone', zone_nick)
+    assert time.get_timezone(db, tmpconfig, None, nick, channel) == zone_nick
```

Comment:
This test is pretty busy, I think it makes sense to fully explode it into its cases, following the lead you've established.

This is a little more complicated-looking, but it does better reflect the complexity of this function.

```suggestion
def test_get_timezone_no_nick_or_channel_data((db: SopelDB, tmpconfig: Config):
    zone = 'Europe/Paris'
    nick = 'IronMan'
    channel = '#test'

    assert time.get_timezone(db) is None
    assert time.get_timezone(db, tmpconfig) == 'UTC', (
        'Must use default timezone from config'
    )
    assert time.get_timezone(db, tmpconfig, zone) == zone


def test_get_timezone_only_nick_data(db: SopelDB, tmpconfig: Config):
    zone_nick = 'America/New_York'
    nick = 'IronMan'
    channel = '#test'
    
    db.set_nick_value(nick, 'timezone', zone_nick)

    assert time.get_timezone(db, tmpconfig, None, nick, channel) == zone_nick
    

def test_get_timezone_only_channel_data(db: SopelDB, tmpconfig: Config):
    zone_channel = 'Asia/Tokyo'
    nick = 'IronMan'
    channel = '#test'

    db.set_channel_value(channel, 'timezone', zone_channel)
    
    assert time.get_timezone(
        db, tmpconfig, None, nick, channel
    ) == zone_channel
    assert time.get_timezone(
        db, tmpconfig, 'Invalid/Tz', nick, channel
    ) == zone_channel,  'Expected channel timezone'


def test_get_timezone_nick_and_channel_data(db: SopelDB, tmpconfig: Config):
    zone_nick = 'America/New_York'
    zone_channel = 'Asia/Tokyo'
    nick = 'IronMan'
    channel = '#test'

    db.set_nick_value(nick, 'timezone', zone_nick)
    db.set_channel_value(channel, 'timezone', zone_channel)
    
    assert time.get_timezone(db, tmpconfig, None, nick, channel) == zone_nick, (
        'Nick timezone must take priority'
    )
```

Note: I folded the removed comments into the corresponding assertion failure messages